modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

```diff
@@ -51,6 +52,10 @@ protected void configure() {
 
         withEnv("SERVICES", services.stream().map(Service::getLocalStackName).collect(Collectors.joining(",")));
 
+        if (hostnameExternal != null) {
+            withEnv("HOSTNAME_EXTERNAL", hostnameExternal);
+        }
```

Comment:
FYI @bsideup, @CauchyPeano, @kiview I've removed the else clause that would set `HOSTNAME_EXTERNAL` by default.

Reply:
I'd like to progress this further and have been thinking about how to do it.

We could leave as-is, but the downside is that everyone using SQS has to remember to use `withHostnameExternal( DockerClientFactory.instance().dockerHostIpAddress())`. That's an OK workaround, but I'd prefer to go with something more ergonomic...

So how about this:
* If `withHostnameExternal` or the env var has been set, always respect that
* Otherwise:
	* if the container is not connected to a `Network` and doesn't have a network alias, then use `dockerHostIpAddress`
	* if the container is connected to a `Network` and has at least one network alias, then use the first network alias

I think that this would work nicely out of the box.

And it should also should be backward compatible with anyone currently using `HOSTNAME_EXTERNAL` manually, which is the only way that anyone can currently be using SQS in either mode. 

FWIW, it's already impossible to use the SQS mock from both the host and over a docker network because `HOSTNAME_EXTERNAL` can only take one value. The above approach doesn't help in that situation, but at least doesn't make it worse.

WDYT @bsideup?

Reply:
that might work! 👍

We could improve it later and start a proxy on localhost (so that `HOSTNAME_EXTERNAL` is always `localhost`) but that will require a bit of work and can be done later if the proposed solution will cause any problems

Reply:
Yeah, I was thinking about that too, but yes we have the potential to add that later should the need arise.

Cheers - I'll implement tonight.

Reply:
I've implemented, with one slight difference: it is taking the _last_ network alias, rather than the first. I've done this because:

* we have a default `tc-*****` alias generated for every container, and users can add an arbitrary number of their own aliases
* it's fiddly, and creates implicit coupling across classes, to try and pick 'the first alias that is not the `tc-*****` one'.
* given that *all we really need is for the behaviour to be predictable and reasonable*, it seems like picking the last is meets that criteria just as well as picking the first.